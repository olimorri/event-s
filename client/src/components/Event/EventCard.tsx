import React from 'react';
import { Box, Flex, Image , Stack, Heading, Text, useColorModeValue } from '@chakra-ui/react';
import { Link as RouterLink } from 'react-router-dom';
import moment from 'moment';
import { motion } from 'framer-motion';


function EventCard ({value}) {

  return (
    <Box
    maxW={'40%'}
    h={'250px'}
    maxH={'50%'}
    w={'25%'}
    bg={useColorModeValue('white', 'gray.900')}
    boxShadow={'2xl'}
    m={2}
    borderRadius="md"
    p={3}
    overflow={'hidden'}>
          <motion.div 
          animate={{
              scale: [1, 0.8, 1],
              // @ts-ignore
              duration: 0.5
            }} 
          transition={{
            duration: 0.5,
            ease: "easeInOut",
            times: [0, 0.2, 0.5, 0.8, 0],
            // loop: Infinity,
            // repeatDelay: 1
          }}
          whileHover={{
            scale: 1.1,
            transition: {
              duration: .2
            }
          }}>
          <RouterLink to={
            {
              pathname: `/events/${value._id}`,
              state: {event: value}}
            }>
              <Flex flexDirection="column" justifyContent="center" alignItems="center">

                <Box
                  h={'20%'}
                  bg={'gray.100'}
                  mt={-6}
                  mx={-6}
                  mb={2}
                  pos={'relative'}
                  >
                    <Image
                      borderRadius="md"
                      src={value.photo}
                      // layout={'fill'}
                      />
                </Box>
                <Stack>
                  <Heading
                    color={useColorModeValue('gray.700', 'white')}
                    fontSize={'medium'}
                    fontFamily={'body'}>
                    {value.name}
                  </Heading>
                  <Text color={'gray.500'}>{value.location}</Text>
                  <Text >{moment(value.date).format("MMM Do YY")}</Text>
                </Stack>

              </Flex>

        </RouterLink>
    </motion.div>
      </Box>
  )
}

export default EventCard;