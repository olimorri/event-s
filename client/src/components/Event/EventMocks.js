const events = [
  {
    limit: 1,
    duration: 1,
    attendees: -2,
    list: ['604a012800da7a8522af8b20'],
    _id: '604a268300da7a8522af8b23',
    name: 'Test Event 1',
    location: 'SW1A 1AA',
    type: 'Food',
    description: 'Test Event 1 description',
    date: '2021-03-20T00:00:00.000Z',
    photo: 'https://source.unsplash.com/random/300x300',
    geolocation: '-0.042843,51.470509',
    owner: '604a0b6300da7a8522af8b21',
    __v: 0,
  },
  {
    limit: 10000000,
    duration: 1,
    attendees: 0,
    list: [],
    _id: '604ce151b61ff67613b5b6d9',
    geolocation: '-0.155083,50.840887',
    owner: '604a0b6300da7a8522af8b21',
    name: 'Test Event 2',
    location: 'SP4 7DE',
    type: 'Other',
    description: 'Test Event 2 description',
    date: '2021-03-20T00:00:00.000Z',
    photo: 'https://source.unsplash.com/random/300x300',
    __v: 0,
  },
  {
    limit: 100,
    duration: 1,
    attendees: 0,
    list: [],
    _id: '604ce171b61ff67613b5b6da',
    geolocation: '-0.155083,50.840887',
    owner: '604a0b6300da7a8522af8b21',
    name: 'Test Event 3',
    location: 'L4 0TH',
    type: 'Sport',
    description: 'Test Event 3 description',
    date: '2021-03-20T00:00:00.000Z',
    photo: 'https://source.unsplash.com/random/300x300',
    __v: 0,
  },
  {
    limit: 12,
    duration: 1,
    attendees: 0,
    list: [],
    _id: '604ce3d6b61ff67613b5b6db',
    geolocation: '0.110623,50.767652',
    owner: '604a0b6300da7a8522af8b21',
    name: 'Test Event 4',
    location: 'SW1A 2AA',
    type: 'Food',
    description: 'Test Event 4 description',
    date: '2021-04-03T00:00:00.000Z',
    photo: 'https://source.unsplash.com/random/300x300',
    __v: 0,
  },
  {
    limit: 12,
    duration: 12,
    attendees: -15,
    list: ['604a012800da7a8522af8b20'],
    _id: '604f8bd4cdb45af7cd1ca265',
    geolocation: '-0.042843,51.470509',
    owner: '604a0b6300da7a8522af8b21',
    name: 'Test Event 5',
    location: 'E20 2ST',
    type: 'Leisure',
    description: 'Test Event 5',
    date: '2021-03-31T00:00:00.000Z',
    photo: 'https://source.unsplash.com/random/300x300',
    __v: 0,
  },
  {
    limit: 221,
    duration: 22,
    attendees: -1,
    list: [],
    _id: '604f8c11cdb45af7cd1ca266',
    geolocation: '0.110623,50.767652',
    owner: '604a0b6300da7a8522af8b21',
    name: 'Test Event 6',
    location: 'EC4M 8AE',
    type: 'Other',
    description: 'Test Event 6 description',
    date: '2021-03-28T00:00:00.000Z',
    photo: 'https://source.unsplash.com/random/300x300',
    __v: 0,
  },
  {
    limit: 2,
    duration: 1,
    attendees: 0,
    list: [],
    _id: '604f8c49cdb45af7cd1ca267',
    geolocation: '-3.597458,50.562947',
    owner: '604a0b6300da7a8522af8b21',
    name: 'Test Event 7',
    location: 'EC3N 4AB',
    type: 'Music',
    description: 'Test Event 7 description',
    date: '2021-03-26T00:00:00.000Z',
    photo: 'https://source.unsplash.com/random/300x300',
    __v: 0,
  },
  {
    limit: 3,
    duration: 1,
    attendees: 0,
    list: [],
    _id: '604fa49085cdd6032dfdf5e0',
    geolocation: '-0.049937,51.470627',
    owner: '604a0b6300da7a8522af8b21',
    name: 'Test Event 8',
    location: 'SE10 0DX',
    type: 'Sport',
    description: 'Test Event 8 description',
    date: '2021-03-31T00:00:00.000Z',
    photo: 'https://source.unsplash.com/random/300x300',
    __v: 0,
  },
];

const event = {
  attendees: 1,
  date: '2021-03-23T00:00:00.000Z',
  description: 'This is a test event',
  duration: 2,
  geolocation: '0.261773,51.116312',
  limit: 2,
  list: ['604a36891b95d669d99ebe2e'],
  location: 'tn25hs',
  name: 'test event',
  owner: '604b5c00ca29f24f751f492c',
  photo: 'https://source.unsplash.com/random/300x300',
  type: 'Food',
  _id: '604b5c00ca29f24f751f492c',
};

module.exports = { events, event };
