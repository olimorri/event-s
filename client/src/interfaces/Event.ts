export interface Event {
  attendees: number;
  date: string;
  description: string;
  duration: number;
  geolocation: string;
  limit: number;
  location: string;
  name: string;
  owner: string;
  photo: string;
  type: string;
  __v?: number;
  _id: string;
}
