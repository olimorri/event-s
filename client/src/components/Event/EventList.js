import React from 'react';
import EventCard from './EventCard';
import { Wrap } from '@chakra-ui/react';

export default function EventList({ value }) {
  return (
    <Wrap w={'100%'} marginTop={5} align="start">
      {value
        .filter((event) => event.date > new Date().toISOString())
        .map((event) => (
          <EventCard value={event} key={event._id} />
        ))}
    </Wrap>
  );
}
