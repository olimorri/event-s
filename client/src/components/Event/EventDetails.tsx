import React from 'react';
import { useParams } from 'react-router-dom';
import Spinner from '../Handling/Spinner';
import { Flex, Stack, Text, Image, Box, Wrap } from '@chakra-ui/react';
import Attend from './Buttons/Attend';
import Unattend from './Buttons/Unattend';
import moment from 'moment';
import { motion } from 'framer-motion';
import {Event} from '../../interfaces/Event'

type EventDetailProps = {
  events: Event[],
  signUpDown: Function,
  user: any
}

export default function EventDetails({ events, signUpDown, user }: EventDetailProps) {
  let { id }: {id: string} = useParams();
  const event: Event | undefined = events.find((el: Event) => el._id === id); //Event: {}
  const handleSubmit = (e: MouseEvent) => {
    e.preventDefault();
    signUpDown('up', id);
  };
  const unattend = (e: MouseEvent) => {
    e.preventDefault();
    signUpDown('down', id);
  };
  console.log(user.eventList);
  return (
    <Flex
      minH={'100vh'}
      justify={'center'}
      bg={'custom.100'}
      bgImage="url('https://res.cloudinary.com/dujun1hoe/image/upload/v1615228154/event-s/gradient-background-26046-26731-hd-wallpapers.jpg_cenrqe.png')"
      bgSize="100% 50%"
      backgroundRepeat="no-repeat"
    >
      {!event ? (
        <Spinner />
      ) : (
        <Box
          w={'60%'}
          minW={'60%'}
          maxW={'60%'}
          bg={'gray.50'}
          marginTop={50}
          borderRadius="md"
          boxShadow={'md'}
          m={20}
          h={'60vw'}
          maxH={'80vw'}
        >
          <motion.div
            initial={{ opacity: 0 }}
            animate={{
              opacity: 1,
              transition: { delay: 0.4 },
            }}
          >
            <Wrap w={'100%'} align={'center'} p={1}>
              <Box w={'50%'} minW={'300px'}>
                <Image
                  borderRadius="md"
                  boxShadow="md"
                  src={event.photo}
                  alt=""
                />
              </Box>
              <Stack w={'50%'} align={'center'}>
                <Flex flexDirection={'column'} justify={'center'}>
                  <Text fontSize="3xl" m={3}>
                    {event.name}
                  </Text>
                  <Text align={'center'} fontSize="xl">
                    {event.location}
                  </Text>
                </Flex>
                {!Object.keys(user).length ||
                user.host ? null : !user.eventList.some(
                    (el: Event) => el._id === id,
                  ) ? (
                  <Attend handleSubmit={handleSubmit} />
                ) : (
                  <Unattend unattend={unattend} />
                )}
              </Stack>
            </Wrap>
            <Box p={4}>
              <Stack
                w={'100%'}
                h={'60vh'}
                direction={{ base: 'column', md: 'row' }}
              >
                <Flex
                  h={'60%'}
                  flex={1}
                  pl={4}
                  align={'end'}
                  justifyContent={'space-evenly'}
                  flexDirection={'column'}
                >
                  <motion.div
                    initial={{
                      opacity: 0,
                      x: '-100vw',
                    }}
                    animate={{
                      opacity: 1,
                      x: 0,
                    }}
                    transition={{
                      // type: 'spring',
                      ease: 'easeIn',
                      delay: 0.5,
                      duration: 0.5,
                    }}
                  >
                    <Wrap align={'center'}>
                      <Image
                        borderRadius="md"
                        src={
                          'https://res.cloudinary.com/dujun1hoe/image/upload/v1615296945/event-s/icons/calendar-2_hmjf1m.png'
                        }
                        alt=""
                      />
                      <Text justify={'center'}>
                        {moment(event.date).format('MMM Do YY')}
                      </Text>
                    </Wrap>
                  </motion.div>
                  <motion.div
                    initial={{
                      opacity: 0,
                      x: '-100vw',
                    }}
                    animate={{
                      opacity: 1,
                      x: 0,
                    }}
                    transition={{
                      // type: 'spring',
                      ease: 'easeIn',
                      delay: 0.6,
                      duration: 0.5,
                    }}
                  >
                    <Wrap align={'center'}>
                      <Image
                        borderRadius="md"
                        src={
                          'https://res.cloudinary.com/dujun1hoe/image/upload/v1615296946/event-s/icons/alarm-2_csdsdf.png'
                        }
                        alt=""
                      />
                      <Text>{event.duration}</Text>
                    </Wrap>
                  </motion.div>
                  <motion.div
                    initial={{
                      opacity: 0,
                      x: '-100vw',
                    }}
                    animate={{
                      opacity: 1,
                      x: 0,
                    }}
                    transition={{
                      // type: 'spring',
                      ease: 'easeIn',
                      delay: 0.7,
                      duration: 0.5,
                    }}
                  >
                    <Wrap align={'center'}>
                      <Image
                        borderRadius="md"
                        src={
                          'https://res.cloudinary.com/dujun1hoe/image/upload/v1615297028/event-s/icons/announcement_pr7pea.png'
                        }
                        alt=""
                      />
                      <Text>{event.limit}</Text>
                    </Wrap>
                  </motion.div>
                  <motion.div
                    initial={{
                      opacity: 0,
                      x: '-100vw',
                    }}
                    animate={{
                      opacity: 1,
                      x: 0,
                    }}
                    transition={{
                      // type: 'spring',
                      // @ts-ignore
                      ease: 'easeIn',
                      delay: 0.8,
                      duration: 0.5,
                    }}
                  >
                    <Wrap align={'center'}>
                      <Image
                        borderRadius="md"
                        src={
                          'https://res.cloudinary.com/dujun1hoe/image/upload/v1615296945/event-s/icons/event_h6dtmv.png'
                        }
                        alt=""
                      />
                      <Text>{event.type}</Text>
                    </Wrap>
                  </motion.div>
                  <motion.div
                    initial={{
                      opacity: 0,
                      x: '-100vw',
                    }}
                    animate={{
                      opacity: 1,
                      x: 0,
                    }}
                    transition={{
                      // type: 'spring',
                      // @ts-ignore
                      ease: 'easeIn',
                      delay: 0.9,
                      duration: 0.5,
                    }}
                  >
                    <Wrap align={'center'}>
                      <Image
                        borderRadius="md"
                        src={
                          'https://res.cloudinary.com/dujun1hoe/image/upload/v1615296945/event-s/icons/done_r4quc3.png'
                        }
                        alt=""
                      />
                      <Text>{event.attendees}</Text>
                    </Wrap>
                  </motion.div>
                </Flex>
                <Flex
                  h={'100%'}
                  flex={2}
                  p={4}
                  alignContent={'center'}
                  flexDirection={'column'}
                >
                  <Text fontWeight="bold" align={'center'}>
                    About this event
                  </Text>
                  <motion.div
                    initial={{
                      opacity: 0,
                      scale: 0,
                    }}
                    animate={{
                      opacity: 1,
                      scale: 1,
                    }}
                    transition={{
                      // type: 'spring',
                      ease: 'easeIn',
                      delay: 1,
                      duration: 1,
                    }}
                  >
                    <Text mt={10} align={'center'}>
                      {event.description.slice(0, 1200)}...
                    </Text>
                  </motion.div>
                </Flex>

                <Stack spacing={4}></Stack>
              </Stack>
            </Box>
          </motion.div>
        </Box>
      )}
    </Flex>
  );
}
