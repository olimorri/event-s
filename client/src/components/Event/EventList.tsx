import React from 'react';
import EventCard from './EventCard';
import { Wrap } from '@chakra-ui/react';
import { EventListProps } from '../../interfaces/EventListProps';
import { Event } from '../../interfaces/Event';

export default function EventList({ value }: EventListProps) {
  return (
    <Wrap w={'100%'} marginTop={5} align="start">
      {value
        .filter((event: Event) => event.date > new Date().toISOString())
        .map((event: Event) => (
          <EventCard value={event} key={event._id} />
        ))}
    </Wrap>
  );
}
