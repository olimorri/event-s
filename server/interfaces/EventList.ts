import { Document } from 'mongoose';
import { IUser } from '../interfaces/User';

export interface IEventList extends Document {
  name: string;
  date?: Date;
  location: string;
  geolocation?: string;
  limit?: number;
  duration?: number;
  attendees?: number;
  photo?: string;
  type: string;
  description?: string;
  owner: string;
  list?: IUser['_id'][];
}
