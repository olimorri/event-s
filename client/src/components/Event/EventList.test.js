import { render, cleanup, screen, fireEvent } from '@testing-library/react';
import { BrowserRouter, Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { events } from '../Event/EventMocks';
import EventList from './EventList';

afterEach(cleanup);

describe('Snapshot', () => {
  it('should take a snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <EventList events={events} />
      </BrowserRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Unit Testing', () => {
  test('renders an event from the list', () => {
    render(
      <BrowserRouter>
        <EventList events={events} />
      </BrowserRouter>,
    );
    const event = screen.getByText(/test event 2/i);
    expect(event).toBeInTheDocument();
  });

  test('does not render a past event', () => {
    render(
      <BrowserRouter>
        <EventList events={events} />
      </BrowserRouter>,
    );
    const event = screen.queryByText(/test event 1/i);
    expect(event).not.toBeInTheDocument();
  });

  test('renders all present/future events', () => {
    render(
      <BrowserRouter>
        <EventList events={events} />
      </BrowserRouter>,
    );
    const allEvents = screen.queryAllByText(/test event/i);
    expect(allEvents).toHaveLength(7);
  });

  test('redirects to correct event details page on event click', () => {
    const history = createMemoryHistory();
    render(
      <Router history={history}>
        <EventList events={events} />
      </Router>,
    );
    const myEvent = screen.getByText(/^Test Event 2$/i);
    expect(history.location).toHaveProperty('pathname', '/');
    expect(history.entries).toHaveLength(1);
    fireEvent.click(myEvent);
    expect(history.location).toHaveProperty(
      'pathname',
      '/events/604ce151b61ff67613b5b6d9',
    );
    expect(history.entries).toHaveLength(2);
    expect(history.entries[1]).toHaveProperty(
      'pathname',
      '/events/604ce151b61ff67613b5b6d9',
    );
  });
});
