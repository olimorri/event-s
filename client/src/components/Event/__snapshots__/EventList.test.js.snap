// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="chakra-wrap css-4hd7e7"
  >
    <ul
      class="chakra-wrap__list css-1mxfdxb"
    >
      <div
        class="css-ujugy8"
      >
        <div>
          <a
            href="/events/604ce151b61ff67613b5b6d9"
          >
            <div
              class="css-u6v8er"
            >
              <div
                class="css-1bj41hx"
              >
                <img
                  class="chakra-image__placeholder css-clx8i5"
                />
              </div>
              <div
                class="chakra-stack css-1pifhv5"
              >
                <h2
                  class="chakra-heading css-y9ag95"
                >
                  Test Event 2
                </h2>
                <p
                  class="chakra-text css-ax9bdp"
                >
                  SP4 7DE
                </p>
                <p
                  class="chakra-text css-0"
                >
                  Mar 16th 21
                </p>
              </div>
            </div>
          </a>
        </div>
      </div>
      <div
        class="css-ujugy8"
      >
        <div>
          <a
            href="/events/604ce171b61ff67613b5b6da"
          >
            <div
              class="css-u6v8er"
            >
              <div
                class="css-1bj41hx"
              >
                <img
                  class="chakra-image__placeholder css-clx8i5"
                />
              </div>
              <div
                class="chakra-stack css-1pifhv5"
              >
                <h2
                  class="chakra-heading css-y9ag95"
                >
                  Test Event 3
                </h2>
                <p
                  class="chakra-text css-ax9bdp"
                >
                  L4 0TH
                </p>
                <p
                  class="chakra-text css-0"
                >
                  Mar 20th 21
                </p>
              </div>
            </div>
          </a>
        </div>
      </div>
      <div
        class="css-ujugy8"
      >
        <div>
          <a
            href="/events/604ce3d6b61ff67613b5b6db"
          >
            <div
              class="css-u6v8er"
            >
              <div
                class="css-1bj41hx"
              >
                <img
                  class="chakra-image__placeholder css-clx8i5"
                />
              </div>
              <div
                class="chakra-stack css-1pifhv5"
              >
                <h2
                  class="chakra-heading css-y9ag95"
                >
                  Test Event 4
                </h2>
                <p
                  class="chakra-text css-ax9bdp"
                >
                  SW1A 2AA
                </p>
                <p
                  class="chakra-text css-0"
                >
                  Apr 3rd 21
                </p>
              </div>
            </div>
          </a>
        </div>
      </div>
      <div
        class="css-ujugy8"
      >
        <div>
          <a
            href="/events/604f8bd4cdb45af7cd1ca265"
          >
            <div
              class="css-u6v8er"
            >
              <div
                class="css-1bj41hx"
              >
                <img
                  class="chakra-image__placeholder css-clx8i5"
                />
              </div>
              <div
                class="chakra-stack css-1pifhv5"
              >
                <h2
                  class="chakra-heading css-y9ag95"
                >
                  Test Event 5
                </h2>
                <p
                  class="chakra-text css-ax9bdp"
                >
                  E20 2ST
                </p>
                <p
                  class="chakra-text css-0"
                >
                  Mar 31st 21
                </p>
              </div>
            </div>
          </a>
        </div>
      </div>
      <div
        class="css-ujugy8"
      >
        <div>
          <a
            href="/events/604f8c11cdb45af7cd1ca266"
          >
            <div
              class="css-u6v8er"
            >
              <div
                class="css-1bj41hx"
              >
                <img
                  class="chakra-image__placeholder css-clx8i5"
                />
              </div>
              <div
                class="chakra-stack css-1pifhv5"
              >
                <h2
                  class="chakra-heading css-y9ag95"
                >
                  Test Event 6
                </h2>
                <p
                  class="chakra-text css-ax9bdp"
                >
                  EC4M 8AE
                </p>
                <p
                  class="chakra-text css-0"
                >
                  Mar 28th 21
                </p>
              </div>
            </div>
          </a>
        </div>
      </div>
      <div
        class="css-ujugy8"
      >
        <div>
          <a
            href="/events/604f8c49cdb45af7cd1ca267"
          >
            <div
              class="css-u6v8er"
            >
              <div
                class="css-1bj41hx"
              >
                <img
                  class="chakra-image__placeholder css-clx8i5"
                />
              </div>
              <div
                class="chakra-stack css-1pifhv5"
              >
                <h2
                  class="chakra-heading css-y9ag95"
                >
                  Test Event 7
                </h2>
                <p
                  class="chakra-text css-ax9bdp"
                >
                  EC3N 4AB
                </p>
                <p
                  class="chakra-text css-0"
                >
                  Mar 26th 21
                </p>
              </div>
            </div>
          </a>
        </div>
      </div>
      <div
        class="css-ujugy8"
      >
        <div>
          <a
            href="/events/604fa49085cdd6032dfdf5e0"
          >
            <div
              class="css-u6v8er"
            >
              <div
                class="css-1bj41hx"
              >
                <img
                  class="chakra-image__placeholder css-clx8i5"
                />
              </div>
              <div
                class="chakra-stack css-1pifhv5"
              >
                <h2
                  class="chakra-heading css-y9ag95"
                >
                  Test Event 8
                </h2>
                <p
                  class="chakra-text css-ax9bdp"
                >
                  SE10 0DX
                </p>
                <p
                  class="chakra-text css-0"
                >
                  Mar 31st 21
                </p>
              </div>
            </div>
          </a>
        </div>
      </div>
    </ul>
  </div>
</DocumentFragment>
`;
